client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you.tsx

```diff
@@ -109,58 +123,66 @@ const MarketplaceThankYou = ( { productSlug }: { productSlug: string } ) => {
 	// Site is already Atomic (or just transferred).
 	// Poll the plugin installation status.
 	useEffect( () => {
-		if ( transferStatus === AtomicTransferComplete && ! pluginOnSite && ! isRequestingPlugins ) {
+		if ( ! siteId || ( ! isJetpackSelfHosted && transferStatus !== transferStates.COMPLETE ) ) {
+			return;
+		}
+
+		// Update the menu after the plugin has been installed, since that might change some menu items.
+		if ( isPluginOnSite ) {
+			dispatch( requestAdminMenu( siteId ) );
+			return;
+		}
+
+		if ( ! isRequestingPlugins ) {
 			waitFor( 1 ).then( () => dispatch( fetchSitePlugins( siteId ) ) );
 		}
-	}, [ isRequestingPlugins, pluginOnSite, dispatch, siteId, transferStatus ] );
+	}, [
+		isRequestingPlugins,
+		isPluginOnSite,
+		dispatch,
+		siteId,
+		transferStatus,
+		isJetpackSelfHosted,
+	] );
 
-	// Update the menu after the site been transferred to Atomic or after the plugin has
-	// been installed, since that might change some menu items
+	// Set progressbar (currentStep) depending on transfer/plugin status.
 	useEffect( () => {
-		dispatch( updateAdminMenuAfterPluginInstallation( siteId, productSlug ) );
+		// We don't want to show the progress bar again.
+		if ( ! showProgressBar ) {
+			return;
+		}
 
-		// Use an empty array of dependencies since we want to run this effect only once.
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [] );
+		setShowProgressBar( ! isPluginOnSite );
 
-	// Set progressbar (currentStep) depending on transfer.status and pluginOnSite.
-	// Step 0 hides the progress bar. It means "complete" or "still loading transfer status".
-	// Steps 1-4 advance through the bar.
-	useEffect( () => {
-		if ( transferStatus === AtomicTransferActive ) {
+		// Sites already transferred to Atomic or self-hosted Jetpack sites only show one step.
+		if ( isJetpack ) {
+			setCurrentStep( 0 );
```

Comment:
On first transfer of a site, I think it might be this conditional that causes a flash of "Step 1 of 4" at the end of the plugin installation process, when the site has fully transferred?

https://user-images.githubusercontent.com/1398304/179801476-72aa1ff6-2c53-40ac-b33d-267069dd7dfe.mov

Reply:
Another good catch! I think I fixed it in 2dd89b6.